src/command.js

```diff
@@ -209,16 +216,48 @@ export default class StorybookCommand extends Command {
           exclude: serializeRegExp(parameters?.percy?.exclude),
           id
         }));
-      });
+      }, WAIT_FOR_TIMEOUT);
     } catch (error) {
       // remove stack traces from any eval error
       if (typeof error === 'string') {
-        throw new Error(error.replace(/^Error:\s(.*?)\n\s{4}at\s.*$/s, '$1'));
+        throw new Error(normalizeEvalError(error));
       } else {
         throw error;
       }
     } finally {
       await page?.close();
     }
   }
+
+  async getStorybookVersion(url) {
+    let version, page;
+    let aboutUrl = new URL('?path=/settings/about', url).href;
+
+    try {
+      page = await this.percy.browser.page();
+      this.log.debug(`Get Storybook version: ${aboutUrl}`);
+
+      await page.goto(aboutUrl);
+      /* istanbul ignore next: no instrumenting injected code */
+      version = await page.eval(async ({ waitFor }, waitForTimeout) => {
+        let headerPath = "//header[starts-with(text(), 'Storybook ')]";
+        let getPath = path => document.evaluate(path, document, null, 9, null).singleNodeValue;
+
+        await waitFor(() => getPath(headerPath), waitForTimeout)
+          .catch(() => Promise.reject(new Error('Failed to find a <header> element')));
+
+        return getPath(headerPath)
+          .innerText
+          .match(/[-]{0,1}[\d]*[.]{0,1}[\d]+/g, '')
+          .join('');
+      }, WAIT_FOR_TIMEOUT);
+    } catch (error) {
+      this.log.debug(`Couldn't retrieve Storybook version: ${normalizeEvalError(error)}`);
```

Comment:
To handle real errors that aren't strings, this can become:

```suggestion
      this.log.debug('Unable to retrieve Storybook version');
      this.log.debug(normalizeEvalError(error));
```

(the `debug` method will handle any real stack traces)